src/Mod/PartDesign/Gui/TaskExtrudeParameters.cpp

```diff
@@ -388,6 +390,94 @@ void TaskExtrudeParameters::addAxisToCombo(App::DocumentObject* linkObj, std::st
         lnk.setValue(linkObj, std::vector<std::string>(1, linkSubname));
 }
 
+void TaskExtrudeParameters::setCheckboxes(int index, bool isPocket)
+{
+    // disable/hide everything unless we are sure we don't need it
+    // exception: the direction parameters are in any case visible
+    bool isLengthEditVisible = false;
+    bool isLengthEdit2Visible = false;
+    bool isOffsetEditVisible = false;
+    bool isOffsetEditEnabled = true;
+    bool isMidplaneEnabled = false;
+    bool isMidplaneVisible = false;
+    bool isReversedEnabled = false;
+    bool isFaceEditEnabled = false;
+
+    TaskPadParameters::Modes modePad;
+    TaskPocketParameters::Modes modePocket;
+
+    if (isPocket)

```

Comment:
TaskExtrudeParameters is the base class of TaskPocketParameters and TaskPadParameters. Now using in a class stuff from derived class is not a great idea.

What you can do instead is moving the enum class Modes of TaskPocketParameters and TaskPadParameters to TaskExtrudeParameters. Now there is one enum entry different of the two classes and you can solve it this way:
```
    enum class Modes {
        Dimension,
        ThroughAll,
        ToLast = ThroughAll,
        ToFirst,
        ToFace,
        TwoDimensions
    };
```

This way you don't need to have two variables modePad & modePocket but only one.

Reply:
This was my first attempt but since the 2 enums are different for Pad and Pocket, I did not find a suitable solution. Through all gets the "through all" length and this is longer than when to the last face. Therefore the 2 enums are different.

Maybe I have a think here? If not, is there another way?

Reply:
> This was my first attempt but since the 2 enums are different for Pad and Pocket, I did not find a suitable solution. Through all gets the "through all" length and this is longer than when to the last face. Therefore the 2 enums are different.

Sorry, but I don't get it. 
Currently we have the two enums
```
    enum class Modes {
        Dimension,
        ToLast,
        ToFirst,
        ToFace,
        TwoDimensions
    };
```
with Dimension=0, ToLast=1, ToFirst=2, ToFace=3, TwoDimensions=4 and
```
    enum class Modes {
        Dimension,
        ThroughAll,
        ToFirst,
        ToFace,
        TwoDimensions
    };
```
with Dimension=0, ThroughAll=1, ToFirst=2, ToFace=3, TwoDimensions=4. The only difference is ThroughAll and ToLast but because both have the same value a combined enum class can be created:
```
    enum class Modes {
        Dimension,
        ThroughAll,
        ToLast = ThroughAll,
        ToFirst,
        ToFace,
        TwoDimensions
    };
```
with Dimension=0, ThroughAll=1, ToLast=1, ToFirst=2, ToFace=3, TwoDimensions=4. So, the important thing is that all enum labels still have the same value and ThroughAll has the same value as ToLast.

Reply:
>>Through all gets the "through all" length and this is longer than when to the last face. Therefore the 2 enums are different.
>>
>> Maybe I have a think here? If not, is there another way?

;-) well, this was already my thinko. That the modes _do_ something different does not mean they need a different index.
So yes, sure, the enum can be sorted out to the helper class too. I updated the PR accordingly and I think the PR should now be ready to merge.